### ISSUE TITLE
Fathom

### ISSUE THREAD

**Issue by ahitabisma** [wealthy-org/Fathom#1]
# Fathom — Technical Documentation & Development Phases

Privacy-First Trust Layer. Skor kredibilitas on-chain yang dibangun murni dari address wallet — tanpa perlu menghubungkan identitas sosial apa pun.

Dokumen ini disusun berdasarkan urutan fase pengembangan, supaya tim dev bisa membangun secara bertahap.

---

## 1. Ringkasan Produk

Fathom adalah trust layer di mana setiap wallet punya **skor kredibilitas** yang terbentuk dari vouch (dukungan berstake token), review, histori aktivitas on-chain, dan badge role yang diverifikasi komunitas — semuanya tanpa mewajibkan user membuka identitas sosial asli (nama, foto, akun Twitter, dsb). Identitas yang tampil ke publik hanya address (dipotong) atau alias buatan sendiri.

**Perbedaan filosofis dari trust-layer sejenis yang sudah besar di pasar**: mereka membangun kredibilitas dari identitas sosial publik (integrasi ke Twitter, feed sosial ramai). Fathom membangun kredibilitas murni dari perilaku on-chain address, cocok untuk user crypto-native yang sengaja tidak ingin menyambungkan reputasi finansial ke identitas dunia nyata.

**Target user**: wallet/komunitas yang butuh cara menilai "bisa dipercaya atau tidak" pihak lain tanpa saling kenal identitas asli — misal sebelum OTC trade anonim, seleksi member komunitas/DAO, atau kolaborasi lintas wallet.

---

## 2. Arsitektur Umum

```text
Wallet connect (Phantom EVM + SIWE)
   │
   ▼
Profil Fathom (address-only / alias) ── Vouch/Review dari wallet lain ──▶ Credibility Score Engine
   │                                                                          │
   ▼                                                                          ▼
Role Badge (self-declare + community attest)                    Dispute/Slash-lite (threshold-report)
   │                                                                          │
   ▼                                                                          ▼
Share Card (@vercel/og) ──▶ dibagikan manual ke sosial            Skor dibekukan sampai ditinjau
```

Tech stack: **TypeScript penuh** — Next.js (App Router) + TypeScript untuk frontend/API, `wagmi` + `viem` untuk wallet & kontrak, Foundry (Solidity + test suite TypeScript-compatible via `forge` scripting) untuk smart contract.

---

## FASE 1 — Profil Wallet-Only (Privacy-First Identity)

Tujuan: identitas dasar berjalan, TANPA skor/vouch dulu.

### 1.1 Connect Wallet & SIWE
Phantom Wallet mode EVM (`wagmi` + `viem`), target Robinhood Chain testnet. Setelah connect, wallet menandatangani pesan SIWE untuk membuktikan kepemilikan sebelum sesi dianggap sah.

### 1.2 Profil Address-Only
Tampilan publik profil HANYA menunjukkan address (dipotong, misal `0x71...4a3f`) — tidak ada kolom wajib nama asli/foto/link sosial. User boleh isi **alias buatan sendiri** (bebas, tidak diverifikasi ke identitas asli mana pun), murni kosmetik.

### 1.3 Starting Baseline dari Aktivitas On-Chain
Sebelum ada vouch/review sama sekali, profil baru tetap punya baseline skor kecil dari sinyal on-chain objektif (umur wallet, jumlah transaksi wajar) — supaya tidak semua wallet baru mulai dari nol mutlak.

**Output Fase 1**: user bisa connect wallet dan punya profil dasar, belum ada mekanisme trust sama sekali.

---

## FASE 2 — Credibility Score Engine

Tujuan: formula skor jelas dan bisa dihitung ulang kapan saja dari data mentah (bukan angka yang disimpan statis tanpa cara reproduksi).

### 2.1 Komponen Formula (Contoh Bobot Awal, Bisa Disesuaikan Saat Tuning)
```
skor = baseline_onchain (maks 100)
     + total_vouch_weighted (maks 400)
     + rata_rata_review * 20 (maks 100, dari skala rating 1-5)
     + bonus_invite (0 atau 50, lihat Fase 6)
     + bonus_role_badge_attested (0 atau 50 per badge, maks 2 badge dihitung)
     - penalti_dispute_aktif (potong 50% dari total sementara, lihat Fase 5)
```
`total_vouch_weighted` = jumlah stake token dari seluruh vouch yang diterima, dengan diminishing return (vouch ke-1 dari 1 wallet dihitung penuh, vouch tambahan dari wallet yang sama dihitung makin kecil) — mencegah 1 wallet vouch berkali-kali ke wallet yang sama untuk menggembungkan skor.

### 2.2 Rekalkulasi
Skor dihitung ulang setiap ada event baru (vouch masuk, review masuk, dispute dibuka/ditutup) — bukan proses batch harian, supaya real-time.

**Output Fase 2**: skor bisa dihitung dan diverifikasi ulang dari data mentah kapan saja.

---

## FASE 3 — Vouch (Stake Token)

Tujuan: sinyal kepercayaan terkuat, berbasis stake token (bukan klik gratis).

### 3.1 Alur Vouch
Wallet A vouch ke wallet B dengan sejumlah token (stake, di-lock di smart contract). Vouch ini kontribusi utama ke skor B (§2.1). Wallet A yang aktif vouch dan tidak pernah kena dispute mendapat reward kecil dari pool insentif.

### 3.2 Anti-Sybil
Karena vouch butuh stake (bukan gratis), membuat banyak wallet palsu untuk saling vouch punya biaya nyata — bukan penghalang sempurna, tapi cukup menaikkan ongkos serangan.

**Output Fase 3**: mekanisme vouch berjalan, terhubung ke smart contract.

---

## FASE 4 — Review (Tanpa Stake)

Tujuan: cara membangun reputasi yang lebih ringan, tanpa komitmen token.

### 4.1 Alur Review
Wallet mana pun bisa memberi rating (1-5) + komentar singkat ke wallet lain, tanpa perlu stake. Kontribusi ke skor lebih kecil dibanding vouch (§2.1), sesuai levelnya yang lebih ringan.

### 4.2 Privacy Guard pada Teks Review (Opsional, Ringan)
Karena prinsip produk ini privacy-first, teks review dipindai otomatis (regex + model gratis OpenRouter) untuk mendeteksi kalau user tidak sengaja menempelkan informasi identitas asli (nama lengkap, handle sosial, email) di kolom komentar — kalau terdeteksi, tampilkan peringatan sebelum submit ("Komentar mengandung info yang mengarah ke identitas asli, lanjutkan?"). Ini murni bantuan pencegahan, bukan validasi wajib yang memblokir keras.

**Output Fase 4**: review berjalan, dengan lapisan perlindungan privasi dasar.

---

## FASE 5 — Dispute / Slash-Lite (Threshold-Report)

Tujuan: mekanisme tolak-ukur kalau ada wallet nakal, tanpa perlu sistem voting validator penuh.

### 5.1 Alur Report
Wallet ber-skor tinggi (di atas ambang tertentu) bisa mengajukan report/flag ke wallet lain. Begitu jumlah report dari wallet-wallet berbeda mencapai ambang N, seluruh vouch yang mendukung wallet yang di-report otomatis ditandai **disputed** — skor wallet tersebut dibekukan sebagian (§2.1 penalti) sampai ditinjau.

### 5.2 Peninjauan
Untuk MVP, peninjauan dilakukan manual oleh tim (lewat panel admin sederhana) — bukan voting komunitas penuh seperti mekanisme slash yang lebih kompleks. Keputusan: pulihkan skor (report ditolak) atau kunci sebagian stake vouch yang terbukti mendukung perilaku curang.

**Output Fase 5**: ada mekanisme akuntabilitas dasar, tanpa perlu membangun sistem voting yang berat.

---

## FASE 6 — Invite Soft-Gate

Tujuan: insentif undangan tanpa mengorbankan kemudahan onboarding.

### 6.1 Alur
Wallet baru **tidak wajib** diundang untuk bisa connect dan pakai Fathom. Tapi kalau masuk lewat link undangan dari wallet lain, dapat **bonus starting credibility +50** (lihat §2.1). Inviter tetap punya ikatan tipis ke wallet yang diundang — kalau yang diundang kena dispute (§5.1), inviter mendapat sedikit dampak negatif ke skornya sendiri, mendorong orang untuk tidak sembarangan undang wallet yang tidak dikenal.

**Output Fase 6**: fitur undangan berjalan sebagai insentif, bukan barrier masuk.

---

## FASE 7 — Tip

Tujuan: apresiasi langsung ke review/vouch yang dianggap bermanfaat.

### 7.1 Alur
Tombol "Tip" muncul di tiap review/vouch yang diterima seorang wallet. Tip dikirim dalam token project, langsung ke wallet penerima, dicatat sebagai riwayat transaksi biasa (tidak memengaruhi skor kredibilitas — murni apresiasi, bukan sinyal trust tambahan).

**Output Fase 7**: ada elemen apresiasi sosial ringan tanpa memengaruhi integritas skor.

---

## FASE 8 — Role/Skill Badge (Pengganti "Attest" Sosial)

Tujuan: cara menunjukkan keahlian/peran tanpa membuka identitas sosial asli.

### 8.1 Self-Declare
Wallet bisa mendeklarasikan diri punya role tertentu (contoh: "Builder", "Reviewer Aktif", "Community Mod") — deklarasi awal berstatus "belum diverifikasi".

### 8.2 Community Attestation
Wallet lain (dengan syarat skor kredibilitas minimum) bisa memberi attestation ke role yang dideklarasikan — kalau attestation dari wallet berbeda mencapai ambang tertentu, badge berubah status jadi "terverifikasi komunitas". Ini menggantikan fitur "attest ke akun sosial eksternal" dari reference — tetap ada proses verifikasi, tapi obyeknya peran/keahlian, bukan identitas dunia nyata.

**Output Fase 8**: ada sistem badge kredibel tanpa perlu link akun sosial.

---

## FASE 9 — Share Card (Pengganti Feed Sosial)

Tujuan: elemen viral/shareable tanpa perlu membangun feed sosial penuh.

### 9.1 Trigger
Setiap kali skor naik level, dapat vouch baru, atau badge terverifikasi, sistem generate 1 gambar share otomatis.

### 9.2 Implementasi
```
GET /api/share/:walletAddress
```
Menggunakan **`@vercel/og`** — pola yang sama seperti dipakai di project Stride. User men-download/share gambar ini secara manual ke sosial media mereka sendiri; Fathom tidak membangun feed publik sendiri.

**Output Fase 9**: ada elemen viral loop ringan, tanpa beban infrastruktur feed real-time.

---

## FASE 10 — Tren Skor (Pengganti Sentiment Tracking Token)

Tujuan: menunjukkan riwayat perkembangan reputasi seseorang, tanpa masuk ke ranah trading/sentimen token.

### 10.1 Alur
Tiap kali skor dihitung ulang (§2.2), simpan snapshot angka + timestamp. Profil menampilkan grafik sparkline sederhana (naik/turun) dari histori skor tersebut — murni tentang reputasi personal, bukan harga/sentimen token.

**Output Fase 10**: user bisa lihat tren reputasinya sendiri maupun wallet lain dari waktu ke waktu.

---

## FASE 11 — Endorsement Directory (Pengganti Trade/Broker)

Tujuan: kegunaan discovery, tanpa masuk ke ranah trading/marketplace.

### 11.1 Alur
Direktori publik yang menampilkan wallet-wallet dengan skor tinggi, dikelompokkan per role badge terverifikasi (§8.2) — contoh: daftar "Builder Terpercaya", "Reviewer Aktif Terverifikasi". Murni referensi/discovery, tidak ada transaksi jual-beli jasa di dalamnya.

**Output Fase 11**: ada nilai kegunaan praktis (menemukan wallet terpercaya per kategori) tanpa membangun fitur trading.

---

## FASE 12 — Score-Gated Access (Pengganti Exclusive Access Berbayar)

Tujuan: skor kredibilitas punya kegunaan nyata di dalam ekosistem sendiri.

### 12.1 Alur
Konten/fitur/akses komunitas tertentu di platform kita bisa disyaratkan skor kredibilitas minimum untuk dibuka — bukan dibeli dengan uang/token secara langsung, tapi "dipantaskan" lewat reputasi yang sudah dibangun.

**Output Fase 12**: skor kredibilitas jadi punya fungsi konkret, bukan cuma angka pajangan.

---

## FASE 13 — Seed Data untuk Pengujian Pra-Rilis

Tujuan: sebelum rilis, tim dev butuh data uji yang mencakup SEMUA fitur di atas supaya QA menyeluruh bisa dilakukan tanpa harus membangun skenario dari nol secara manual satu-satu.

### 13.1 Ketentuan Seed Data
Siapkan **minimal 30 akun wallet** (bisa wallet testnet yang di-generate lokal, tidak perlu wallet asli), dengan kombinasi skenario berikut (boleh 1 wallet memenuhi lebih dari 1 skenario sekaligus supaya efisien mencapai 30 akun):

| Skenario | Jumlah Minimal Wallet | Tujuan Pengujian |
|---|---|---|
| Skor tinggi, banyak vouch diterima | 5 | Uji tampilan profil skor tinggi, kontribusi vouch ke formula |
| Skor rendah/baru (belum ada vouch/review) | 5 | Uji baseline on-chain (§1.3), tampilan cold-start |
| Masuk lewat invite (dapat bonus) vs tidak | 3 vs 3 | Uji bonus starting credibility (§6.1) |
| Sedang dalam status disputed/flagged | 3 | Uji pembekuan skor & tampilan status dispute (§5.1) |
| Pernah menerima tip di review/vouch-nya | 3 | Uji alur tip (§7.1) |
| Role badge self-declared belum diverifikasi | 3 | Uji tampilan badge status "belum diverifikasi" (§8.1) |
| Role badge sudah terverifikasi komunitas | 3 | Uji tampilan badge "terverifikasi" + masuk Endorsement Directory (§8.2, §11.1) |
| Punya histori tren skor (naik & turun) | 3 | Uji grafik sparkline (§10.1) |
| Skor di atas ambang score-gated access | 2 | Uji akses konten ter-gate berhasil terbuka (§12.1) |
| Skor di bawah ambang score-gated access | 2 | Uji akses konten ter-gate berhasil tertutup |
| Sudah pernah generate share card | 2 | Uji `@vercel/og` share card (§9.1) |

Total unik minimal 30 wallet (dengan overlap antar skenario diperbolehkan). Seed data ini disiapkan lewat script (`scripts/seed.ts`, TypeScript) yang dijalankan sekali sebelum rilis ke tim dev untuk keperluan QA — bukan data yang ikut dideploy ke environment produksi.

**Output Fase 13**: tim dev punya data uji lengkap yang mencakup seluruh fitur, siap dipakai QA sebelum rilis publik.

---

### 14.1 Smart Contract (Robinhood Chain)
| Contract | Fungsi |
|---|---|
| `FathomVouchRegistry` | Mencatat vouch (stake, pemberi, penerima), dan status dispute/slash-lite |
| `FathomInviteRegistry` | Mencatat hubungan invite antar wallet, dipakai untuk bonus starting credibility & dampak dispute ke inviter |

Deploy ke testnet Robinhood Chain (Chain ID `46630`) terlebih dahulu, uji penuh seluruh Fase 1–13, baru lanjut ke mainnet (Chain ID `4663`).

### 14.2 Checklist Sebelum Rilis
- [ ] Seluruh 14 fase sudah diuji menggunakan seed data Fase 13.
- [ ] Formula skor (§2.1) sudah divalidasi tidak mudah digembungkan (uji kasus 1 wallet vouch berkali-kali ke wallet yang sama).
- [ ] Privacy Guard (§4.2) berhasil mendeteksi contoh teks yang mengandung info identitas asli.
- [ ] Dispute/slash-lite (§5.1) teruji dari alur report sampai skor dibekukan dan dipulihkan.
- [ ] Tidak ada field wajib di profil yang mengekspos identitas asli user (§1.2).

---

## Kebutuhan Environment Variable / API

Karena tidak ada fitur trading/companion yang butuh LLM berat, kebutuhan API eksternal di project ini minim:

```bash
# Wallet & Chain
NEXT_PUBLIC_ROBINHOOD_CHAIN_ID=46630
NEXT_PUBLIC_ROBINHOOD_RPC_URL=https://rpc.testnet.chain.robinhood.com
NEXT_PUBLIC_TOKEN_CONTRACT_ADDRESS=
NEXT_PUBLIC_FATHOM_VOUCH_REGISTRY_ADDRESS=
NEXT_PUBLIC_FATHOM_INVITE_REGISTRY_ADDRESS=

# Privacy Guard pada Review (Fase 4.2) — opsional, hanya jika fitur ini diaktifkan
OPENROUTER_API_KEY=
OPENROUTER_MODEL=            # pakai model gratis (:free)

# Cron eksternal (dispute timeout otomatis, snapshot tren skor harian — Fase 10 & 5)
CRON_SECRET=
```

**Catatan**: `OPENROUTER_API_KEY` hanya dipakai untuk fitur Privacy Guard (§4.2) yang sifatnya bantuan pencegahan, bukan komponen inti — kalau mau MVP paling minim, fitur ini bisa dinonaktifkan dulu tanpa mengganggu fitur lain. RPC Robinhood Chain pakai endpoint publik gratis, tidak butuh API key berbayar.

---

## Hal yang Masih Perlu Diputuskan Sebelum Development Dimulai

- **Ambang angka pasti** untuk: jumlah report minimum memicu dispute (§5.1), skor minimum untuk bisa memberi attestation badge (§8.2), skor minimum untuk score-gated access (§12.1). Formula di §2.1 memberi kerangka, tapi angka final sebaiknya diuji dulu dengan seed data Fase 13 sebelum dikunci.
- **Sumber baseline on-chain** (§1.3): perlu ditentukan indikator konkret apa saja yang dipakai (umur wallet sejak transaksi pertama, jumlah transaksi minimum, dll) dan bobotnya.
- **Kebijakan token reward** untuk voucher aktif (§3.1): dari pool mana, berapa besar, dan skema distribusinya — belum didetailkan di dokumen ini karena bersinggungan dengan tokenomics keseluruhan project.
